### PR TITLE
Feature create basic gem functionality

### PR DESCRIPTION
**Pull request by janeskim**

### CODE REVIEW

File: spec/humanity_api_spec.rb [hunk 1]
```diff
@@ -0,0 +1,180 @@
+require "spec_helper"
+
+describe HumanityApi do
+  let(:humanity_api) { HumanityApi.new }
+
+  let(:key) { '9c5edef7302e2f272f8de273b1d20927d65b1c49' }
```
Comment (joshbeckman) [inline spec/humanity_api_spec.rb]: Can we use a dummy key here? Or use `vcr` to play static responses? Or create an unimportant key?

File: lib/humanity_api/humanity_api.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    method_error = "Invalid data packet: please provide a valid request method of 'GET', 'CREATE', 'UPDATE', or 'DELETE'."
+    module_error = "Invalid data packet: please provide the Humanity module." 
+
+    !body[:key] ? @errors << key_error : @errors.delete(key_error)
+    !VALID_REQUEST_METHODS.include?(body[:request][:method]) ? @errors << method_error : @errors.delete(method_error)
+    !body[:request][:module] ? @errors << module_error : @errors.delete(module_error)
+  end
+
+  def request_humanity_data(data)
+    request_params = build_request_params(data)
```
Comment (janeskim) [inline lib/humanity_api/humanity_api.rb:29]: initialize empty @errors

File: lib/humanity_api/humanity_api.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    @output_format = "json"
+  end
+
+  def errors?
+    @errors.count > 0
+  end
+
+  def check_params_for_errors(body)
+    key_error = "Invalid data packet: please provide a Humanity API key."
+    method_error = "Invalid data packet: please provide a valid request method of 'GET', 'CREATE', 'UPDATE', or 'DELETE'."
```
Comment (janeskim) [inline lib/humanity_api/humanity_api.rb]: take out this check

File: lib/humanity_api/humanity_api.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    request_params = JSON.generate(body)
+    request.set_form_data({"data" => request_params})
+    response = http.request(request)
+
+    case response.code.to_i
+    when 200 || 201
+      JSON.parse(response.body)
+    else
+      @errors << "Request failed: #{response.value}"
+      JSON.parse(response) # think about when this will return falsey/nil/error
```
Comment (janeskim) [inline lib/humanity_api/humanity_api.rb]: have begin catch so that if json is not parsable, return an error

File: spec/humanity_api_spec.rb (hunk 1 above)
Comment (janeskim) [reply spec/humanity_api_spec.rb]: i'll use a dummy key

File: lib/humanity_api/humanity_api.rb [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    end
+  end
+
+  def parse_humanity_response(response)
+    if errors?
+      { errors: @errors.join(" ") }
```
Comment (janeskim) [inline lib/humanity_api/humanity_api.rb]: return this as `error` to be consistent with humanity's response (like response['error']

File: spec/humanity_api_spec.rb [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+  describe '#check_params_for_errors' do
+    context "given that an invalid data packet is passed in" do
+      it "adds an error message for a missing key" do
+        invalid_request_params.delete(:key)
+        request_params = humanity_api.check_params_for_errors(invalid_request_params)
+        expect(humanity_api.errors).to include("Invalid data packet: please provide a Humanity API key.")
+      end
+
+      it "adds an error message for an invalid request method" do
```
Comment (janeskim) [inline spec/humanity_api_spec.rb]: take this out

File: README.md [hunk 7]
```diff
@@ -1 +1,71 @@
...
+    $ gem install humanity_api
+
+## Usage
+
+```ruby
+
+# Use this gem by instantiating the class.
+humanity_api = HumanityApi.new
+
+# If you have the key and token set in your environment variables, the gem will find and use them. You can also access the key and token using getters and setters:
```
Comment (janeskim) [inline README.md]: specify which environment variables "HUMANITY_TOKEN" ..

File: lib/humanity_api/humanity_api.rb [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    response = http.request(request)
+
+    case response.code.to_i
+    when 200 || 201
+      JSON.parse(response.body)
+    else
+      @errors << "Request failed: #{response.value}"
+      begin
+        JSON.parse(response)
+      rescue
```
Comment (janeskim) [inline lib/humanity_api/humanity_api.rb]:
``` ruby
rescue error
  { error: error }
end
```

**Pull request merged**